pydeseq2/utils.py

```diff
@@ -170,14 +170,33 @@ def build_design_matrix(
         A DataFrame with experiment design information (to split cohorts).
         Indexed by sample barcodes.
     """
+
     design_matrix = pd.get_dummies(clinical_df[design])
+
+    if design_matrix.shape[-1] == 1:
+        raise ValueError(
+            f"The design factor takes only one value: "
+            f"{design_matrix.columns.values.tolist()}, but two are necessary."
+        )
+    elif design_matrix.shape[-1] > 2:
+        raise ValueError(
+            f"The design factor takes more than two values: "
+            f"{design_matrix.columns.values.tolist()}, but should have exactly two."
+        )
```

Comment:
Could the tests be merged ? using != 2, f'the design factor takes {len(design_matrix.shape[-1])} values: {design_matrix.columns.values.tolist()}, but should have exactly 2'

Reply:
I don't know, it seems more informative this way? Although it would save a couple of code lines indeed

Reply:
From my point of view printing "but two are necessary." is a less informative compare to "but should have exactly two.", so I would copy-past the sentence, and thus the tests. Refactoring would make the code clearer I think. But it is not important so I would approve even without this change